### PR TITLE
chore: validate events from the sdk in api gateway

### PR DESCRIPTION
**Pull request by nnnkkk7**
- Reverts bucketeer-io/bucketeer#1637
- Validate if userID is empty or userID in User Object is empty.(We want to use User Object, but userID is still used in old sdk)
- Unify userID retrieval in subscriver because some processers use userID not in User Object.

### CODE REVIEW

File: pkg/api/api/grpc_validation.go [hunk 1]
```diff
@@ -105,6 +110,30 @@ func (v *eventGoalValidator) validate(ctx context.Context) (string, error) {
...
+		v.logger.Debug(
+			"Empty goal_id",
+			log.FieldsFromImcomingContext(ctx).AddFields(
+				zap.String("id", v.event.Id),
+				zap.String("goal_id", ev.GoalId),
+			)...,
+		)
+		return codeEmptyField, errEmptyGoalID
+	}
+	if ev.User.Id == "" && ev.UserId == "" {
```
Comment (cre8ivejp) [inline pkg/api/api/grpc_validation.go:125]: The `ev.User` could be nil, so we must check it.

File: pkg/api/api/grpc_validation.go [hunk 2]
```diff
@@ -148,6 +177,50 @@ func (v *eventEvaluationValidator) validate(ctx context.Context) (string, error)
...
+		v.logger.Debug(
+			"Empty variation_id",
+			log.FieldsFromImcomingContext(ctx).AddFields(
+				zap.String("id", v.event.Id),
+				zap.String("variation_id", ev.VariationId),
+			)...,
+		)
+		return codeEmptyField, errEmptyVariationID
+	}
+	if ev.User.Id == "" && ev.UserId == "" {
```
Comment (cre8ivejp) [inline pkg/api/api/grpc_validation.go:202]: Same here.

File: pkg/api/api/grpc_validation.go (hunk 1 above)
Comment (nnnkkk7) [reply pkg/api/api/grpc_validation.go]: Thank you!
I fixed it!
[fix: add nil check for User object](https://github.com/bucketeer-io/bucketeer/pull/1650/commits/e06e1f1b6d99c9de003c2da605050d0abc61d0db)

Comment (cre8ivejp) [reply pkg/api/api/grpc_validation.go]: Thank you!
Just to be safe, can you add a unit test for nil case?

Comment (nnnkkk7) [reply pkg/api/api/grpc_validation.go]: ok!
I added it!
[fix: add test case for nil check](https://github.com/bucketeer-io/bucketeer/pull/1650/commits/455db9ad6ba00586c2980efae0f19e291e7b850f)

**Review by cre8ivejp — APPROVED** [review]
Nice!
Thank you 🎉
